### PR TITLE
Dotnet install-script updates

### PR DESCRIPTION
**Pull request by ghost**
Introduce support for arm & arm64 architectures; when detecting a missing 'library' prerequisite - do not fail the installation.

Addresses: https://github.com/dotnet/cli/issues/9007

**Auto-detect tested successfully on:**
OS = raspbian.8
Arch = armv7l

### CODE REVIEW

**Comment by natemcmaster** [comment]
Any reason we can't use `uname -m` to detect arch? Here are some sample values. I'm sure there are more. We could always err if we don't recognize the arch.

Platform | `uname -m` | Rid
---|---|---
macOS | `x86_64` | `osx-x64`
Ubuntu Bionic x64 | `x86_64` | `linux-x64`
Ubuntu Bionic arm32v7 | `armv7l` | `linux-arm`
Ubuntu Bionic arm64v8 | `aarch64` | `linux-arm64`

File: scripts/obtain/dotnet-install.sh [hunk 1]
```diff
@@ -309,32 +303,24 @@ combine_paths() {
...
-            return 0
-            ;;
         amd64|x64)
             echo "x64"
             return 0
             ;;
-        x86)
-            say_err "Architecture \`x86\` currently not supported"
-            return 1
+         arm|arm32)
```
Comment (janvorli) [inline scripts/obtain/dotnet-install.sh]: Do we need to add the "arm32"? We don't use that name anywhere, I believe.

File: scripts/obtain/dotnet-install.sh [hunk 2]
```diff
@@ -790,7 +776,7 @@ temporary_file_template="${TMPDIR:-/tmp}/dotnet.XXXXXXXXX"
 channel="LTS"
 version="Latest"
 install_dir="<auto>"
-architecture="<auto>"
+architecture="x64"
```
Comment (janvorli) [inline scripts/obtain/dotnet-install.sh]: A nit - should the default be the architecture of the current platform? the 'uname -m' returns a Linux specific name, so we can do:
```sh
CPUName=$(uname -m)
case $CPUName in
i686)
    architecture="x86"
    ;;
x86_64|amd64)
    architecture="x64"
    ;;
armv7l)
    architecture="arm"
    ;;
aarch64)
    architecture="arm64"
    ;;
esac
```

**Comment by johnbeisner** [comment]
@natemcmaster 
@janvorli 
In the current version; the default architecture is 'x64' when "<auto>" is used.

I can see the value in trying to recognize the architecture; however, I feel strongly that we need to default to 'x64' when "<auto>" is used instead of an error-out. We would, at minimum, risk breaking fringe x64 Linux platforms if we now error-out on a peculiar return-value from 'uname -m'.

**Review by janvorli — APPROVED** [review]

**Review by natemcmaster — APPROVED** [review]

**Comment by johnbeisner** [comment]
@dotnet-bot Test RHEL7.2 x64 Release Build

**Pull request merged**